### PR TITLE
Use `residentKey` instead of `requireResidentKey`

### LINKED ISSUE #73: [BUG] - `requireResidentKey` is deprecated. Replace with `residentKey`.
https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredentialCreationOptions 

Testing on various devices, the passkey was not saving.

Problem fixed by changing `requireResidentKey: true` to `residentKey: 'required'`.

I will fix this one too.

### PR DESCRIPTION
**Pull request by peaceful-james**
Resolves #73

### CODE REVIEW

**Comment by peaceful-james** [comment]
It just occurred to me that perhaps we should provide both keys, in case the deprecated `requireResidentKey` is still used somewhere.

File: lib/webauthn_components/registration_component.ex [hunk 1]
```diff
@@ -139,7 +133,8 @@ defmodule WebauthnComponents.RegistrationComponent do
       challenge: Base.encode64(challenge.bytes, padding: false),
       excludeCredentials: [],
       id: id,
-      require_resident_key: require_resident_key,
+      residentKey: resident_key,
+      requireResidentKey: resident_key in [:required, :preferred],
```
Comment (type1fool) [inline lib/webauthn_components/registration_component.ex:137]: Should this be `true` when resident key is `:preferred`?

> The value should be set to true if residentKey is set to "required".
> https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredentialCreationOptions#requireresidentkey

**Comment by peaceful-james** [comment]
@type1fool Please merge this one first. I rebased the other 2 off this one so their diffs include this work.

**Review by type1fool — APPROVED** [review]

**Comment by type1fool** [comment]
Good stuff @peaceful-james!

**Pull request merged**